openassessment/workflow/workflow_batch_update_api.py

```diff
@@ -282,126 +285,163 @@ def get_workflow_update_data(peer_workflows):
     {
       "courses": [
         {
-          "course_id": "value",
+          "course_id": str,
           "course_settings": {},
           "assessments": [
             {
-              "item_id": "value",
+              "item_id": str,
               "assessment_requirements": {},
-              "submissions": [
-                {
-                  "submission_uuid": "value"
-                }
-              ]
+              "submissions": [str]
             }
           ]
         }
       ]
     }
     ```
     """
+
     workflow_update_data = {}
     store = modulestore()
-    cache = set([])
+    # temp cache to optimize number of DB lookups for course blocks
+    course_settings_cache = {}
+    # temp cache to optimize number of DB lookups for ora blocks
+    assessment_requirements_cache = {}
+
+    submissions_cache = set([])
+
     for peer_workflow in peer_workflows:
+
         try:
-            if peer_workflow.course_id not in cache:
-                # retrieve course block
+            # pylint: disable=consider-iterating-dictionary
+            if peer_workflow.course_id not in course_settings_cache.keys():
```

Comment:
The call to keys() and the pylint are not necessary. `if <value> in <dict>:` checks if the value is in the key set implicitly.